network.tf

```diff
@@ -0,0 +1,48 @@
+resource "azurerm_virtual_network" "demo" {
+  name                = "${var.prefix}-${var.env_type}-network"
+  location            = var.region
+  resource_group_name = azurerm_resource_group.rg.name
+  address_space       = ["10.0.0.0/16"]
```

Comment:
all necessary statics should be kept in the variables or in the locals variables, additionally for network calculations you can use network functions from terraform: https://www.terraform.io/language/functions/cidrsubnet